packages/terra-site/src/ThemeProvider.jsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  /**
+   * The component(s) that will be wrapped by <ThemeProvider />
+   */
+  children: PropTypes.node.isRequired,
+  /**
+   * Name of class for specified theme. e.g `terra-OSC-theme`
+   */
+  themeName: PropTypes.string.isRequired,
+};
+
+const ThemeProvider = ({ themeName, children, ...customProps }) => {
+  return <div className={themeName}>{children}</div>;
```

Comment:
I created a `<ThemeProvider />` and a `<ThemeProviderChildren />` component. Inside of the `<ThemeProviderChildren />` I used ImmutableJS's is function to compare `props.children`, e.g. 
```js
shouldComponentUpdate(nextProps) {
    // performs a deep equality check if necessary
    return is(this.props.children, nextProps.children);
 }
```
This works well, but I found changeing routes via react-router / react-router's link component doesn't trigger shouldComponentUpdate to return true. There's a way to check routes with previous route [documented here](https://stackoverflow.com/questions/30078260/react-js-shouldcomponentupdate-and-react-router-link) but I'm thinking for now, leaving out the `shouldComponentUpdate` check to keep it simple and compatible with react-router.